### PR TITLE
Users/sogangul/bicep add multi repo solution - Publish Wiki Guidelines for Multi-repo Orchestration with CARML

### PR DESCRIPTION
**Pull request by SomilGanguly**
# Change

***Feel free to remove this sample text***
>Thank you for your contribution !

Publishing wiki guidelines to use Multi-Repo Orchestration with Carml Modules

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Review by MrMCake — CHANGES_REQUESTED** [review]
Let's have a discussion about that one some time. There are a few points I'd like to discuss

File: docs/wiki/ModulesUsage.md [hunk 1]
```diff
@@ -15,9 +15,16 @@ This section gives you an overview of how to use the bicep modules.
     - [**Remote:** Azure CLI](#remote-azure-cli)
 - [Orchestrate deployment](#orchestrate-deployment)
   - [Template-orchestration](#template-orchestration)
+- [Solution deployment](#solution-deployment)
```
Comment (MrMCake) [inline docs/wiki/ModulesUsage.md:18]: Let's move under `Orchestrate deployment / Pipeline-orchestration / Multi-Repository-Approach`

File: docs/wiki/ModulesUsage.md [hunk 2]
```diff
@@ -290,3 +297,192 @@ resource vnet 'Microsoft.Resources/deployments@2021-01-01' = {
...
+
+> 1. 'Azure/ResourceModules' repo has been checked out at the root location intentionally because the `deep-mm/set-variables@v1.0` task expects the _global.variables.json_ file in the _.github/variables/_ location. The GitHub Actions also expect the underlying utility scripts at a specific location
+> 1. 'contoso/MultiRepoTest' repo has been checked out in a nested folder called as "MultiRepoTestParentFolder" to distinguish it from the folders from the other repo in the agent but can be downloaded at the root location too if desired
+> 1. Comparison between IaCS and CARML -
+>
+>    Snippet of Resource Group deployment job from the _Solution_ repo of IaCS -
+>
+>    ![IaCS_RGDeployJob](/docs/media/IaCS_DeployRGJob.png)
+>
+>    | Sr. no. | Topic| IaCS | CARML |
```
Comment (MrMCake) [inline docs/wiki/ModulesUsage.md:483]: Remove table

File: docs/wiki/ModulesUsage.md [hunk 3]
```diff
@@ -290,3 +297,192 @@ resource vnet 'Microsoft.Resources/deployments@2021-01-01' = {
...
+
+### Sample solution
+
+#### Summary
+
+1. Below is an example which uses a _multi-repo_ approach
+1. It fetches the _public_ **Azure/ResourceModules** repo for consuming bicep modules and uses the parameter files present in the _private_ **Contoso/MultiRepoTest** repo for deploying infrastructure
+1. This example is for creating a network hub with following resources -
+    1. First stage: **Deploy resource group**
+        1. Checkout 'Azure/ResourceModules' repo at root of the agent
```
Comment (MrMCake) [inline docs/wiki/ModulesUsage.md:313]:
```suggestion
        1. Checkout 'Azure/ResourceModules' repo at root of the agents
```

File: docs/wiki/ModulesUsage.md (hunk 1 above)
Comment (kavishshivhare) [reply docs/wiki/ModulesUsage.md]: Moved.

File: docs/wiki/ModulesUsage.md [hunk 4]
```diff
@@ -290,3 +297,192 @@ resource vnet 'Microsoft.Resources/deployments@2021-01-01' = {
...
+
+#### Notes
+
+> 1. 'Azure/ResourceModules' repo has been checked out at the root location intentionally because the `deep-mm/set-variables@v1.0` task expects the _global.variables.json_ file in the _.github/variables/_ location. The GitHub Actions also expect the underlying utility scripts at a specific location
+> 1. 'contoso/MultiRepoTest' repo has been checked out in a nested folder called as "MultiRepoTestParentFolder" to distinguish it from the folders from the other repo in the agent but can be downloaded at the root location too if desired
+> 1. Comparison between IaCS and CARML -
+>
+>    Snippet of Resource Group deployment job from the _Solution_ repo of IaCS -
+>
+>    ![IaCS_RGDeployJob](/docs/media/IaCS_DeployRGJob.png)
```
[the remark this answers is not in the document]
Comment (kavishshivhare) [reply docs/wiki/ModulesUsage.md]: Removed.

File: docs/wiki/ModulesUsage.md (hunk 2 above)
Comment (kavishshivhare) [reply docs/wiki/ModulesUsage.md]: Removed.

**Review by MrMCake — CHANGES_REQUESTED** [review]
Temporarily blocked due to Wiki rework #403 

cc: @eriqua

File: docs/wiki/ModulesUsage.md [hunk 5]
```diff
@@ -307,3 +314,105 @@ The example assumes you are using a [`bicepconfig.json`](https://docs.microsoft.
     }
 }
 ```
+
+## Solution deployment
+
+The bicep modules provided by this repo can also be clubbed together to create more complex infrastructures and in the form a completely reusable solutions or products.
```
Comment (MrMCake) [inline docs/wiki/ModulesUsage.md:320]:
```suggestion
The modules provided by this repo can be orchestrated to create more complex infrastructures and as such reusable solutions or products.
```

File: docs/wiki/ModulesUsage.md [hunk 6]
```diff
@@ -307,3 +314,105 @@ The example assumes you are using a [`bicepconfig.json`](https://docs.microsoft.
     }
 }
 ```
+
+## Solution deployment
```
Comment (MrMCake) [inline docs/wiki/ModulesUsage.md:318]:
```suggestion
## Pipeline-orchestration
```

File: docs/wiki/ModulesUsage.md (hunk 5 above)
Comment (MrMCake) [reply docs/wiki/ModulesUsage.md:320]: Overall, I', not sure if that fit's here. The section should be `Pipeline-orcehstration` and be a counter-part to the explaination what template-orchestration is (above).

Then, there should be a sub-header `# Multi-Repository approach` which explicitely explains this flavor

File: docs/wiki/ModulesUsage.md [hunk 7]
```diff
@@ -307,3 +314,105 @@ The example assumes you are using a [`bicepconfig.json`](https://docs.microsoft.
...
+
+The bicep modules provided by this repo can also be clubbed together to create more complex infrastructures and in the form a completely reusable solutions or products.
+
+### Sample solution
+
+#### Summary
+
+1. Below is an example which uses a _multi-repository_ approach
```
Comment (MrMCake) [inline docs/wiki/ModulesUsage.md:326]:
```suggestion
1. Below you can find an example which uses makes use of multiple repositories to orchestrat the deployment (also known as a _multi-repository_ approach)
```

**Review by MrMCake — CHANGES_REQUESTED** [review]
Alright @kavishshivhare, we should be good to redirect the content to the `Soluction creation.md` page. Can you do that?

File: docs/wiki/Solution creation.md [hunk 8]
```diff
@@ -315,3 +334,105 @@ The example assumes you are using a [`bicepconfig.json`](https://docs.microsoft.
     }
 }
 ```
+
+# Pipeline-orchestration
+
+The modules provided by this repo can be orchestrated to create more complex infrastructures and as such reusable solutions or products.
```
Comment (MrMCake) [inline docs/wiki/Solution creation.md:340]: We should a few sentences here to explain the idea behind this orchestration option. For example:

```suggestion
The modules provided by this repo can be orchestrated to create more complex infrastructures and as such reusable solutions or products. This approach leverages the main module repository alongside its contained modules & pipeline templates to deploy resources. Each pipeline job deploys one instance of a resources and their order is controlled by specifying dependencies in the pipeline itself. 
```

**Review by MrMCake — APPROVED** [review]

**Pull request merged**